connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -565,6 +565,47 @@ class SparkConnectPlanner(session: SparkSession) {
         val children = fun.getArgumentsList.asScala.toSeq.map(transformExpression)
         Some(In(children.head, children.tail))
 
+      case "___lambda_function___" =>
+        // UnresolvedFunction[___lambda_function___, ["x, y -> x < y", "x", "y"]]
+
+        if (fun.getArgumentsCount < 2) {
+          throw InvalidPlanInput(
+            "LambdaFunction requires at least 2 child expressions: LamdaFunction, Arguments")
+        }
+
+        val children = fun.getArgumentsList.asScala.toSeq.map(transformExpression)
+
+        val function = children.head
+
+        val variableNames = children.tail.map {
+          case variable: UnresolvedAttribute if variable.nameParts.length == 1 =>
+            variable.nameParts.head
+          case other =>
+            throw InvalidPlanInput(
+              "LambdaFunction requires all arguments to be UnresolvedAttribute with " +
+                s"single name part, but got $other")
```

Comment:
There are two interesting issues here:

 1. When someone submits to the API an expression that does not transform into `UnresolvedExpression` this would throw a weird error message about the name parts but actually the type does not match.
 2. Why the restriction to single part names? Is this a Spark limitation?

Reply:
1, the error message also mentioned `UnresolvedAttribute`

2, existing implementation in PySpark only use single part name